hooks/pre/10-reset_feature.rb

```diff
@@ -35,26 +70,95 @@ def reset_candlepin
   commands = [
     'rm -f /var/lib/candlepin/cpdb_done',
     'rm -f /var/lib/candlepin/cpinit_done',
-    "service #{tomcat} stop",
-    'sudo su postgres -c "dropdb candlepin"'
+    "service #{tomcat} stop"
   ]
-
   Kafo::Helpers.execute(commands)
+  empty_candlepin_database
+end
+
+def empty_mongo
+  mongo_config = load_mongo_config
+  if remote_host?(mongo_config[:host])
+    empty_remote_mongo(mongo_config)
+  else
+    Kafo::Helpers.execute(
+      [
+        'service-wait mongod stop',
+        'rm -f /var/lib/mongodb/pulp_database*',
+        'service-wait mongod start'
+      ]
+    )
+  end
+end
+
+def load_mongo_config
+  config = {}
+  seeds = param_value('katello', 'pulp_db_seeds')
```

Comment:
Fixed

Reply:
this is probably wrong, when you have mongo replica set, you should to connect to primary node, because secondary will not allow you to write
```
# mongo --host 192.168.122.93:27017
replica1:SECONDARY> db.dropDatabase()
{ "note" : "from execCommand", "ok" : 0, "errmsg" : "not master" }
```
it cannot be guaranteed that the first one is primary, and the first can be down

you need url like `replica_name/host1:port,host2:port,...` to connect to primary
```
# mongo --host replica1/192.168.122.93:27017,192.168.122.35:27017
replica1:PRIMARY> use rere
switched to db rere
replica1:PRIMARY> db.dropDatabase()
{ "dropped" : "rere", "ok" : 1 }
```

Reply:
@alda519, good point.
 
The question is - do we support replicas, do we want to block this PR on it?

Reply:
I tried to look into this and it will need more investigation. According to some SO [1] comments the `--host` params can differ on 3.2 and 3.4+. I need to get more familiar with the setup and we should decide on which Mongo DB versions we want to support and test. I'll file a bug to track this.

Reply:
http://projects.theforeman.org/issues/23337